File: Patchr/AirTableViewCell.m
//
//  AirTableViewCell.m
//  Patchr
//
//  Created by Jay Massena on 5/8/15.
//  Copyright (c) 2015 3meters. All rights reserved.
//

#import "AirTableViewCell.h"

// From: http://stackoverflow.com/questions/19256996/uibutton-not-showing-highlight-on-tap-in-ios7/26049216#26049216

@implementation AirTableViewCell

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    
    if (self) {
        [self delayContentTouches];
    }
    return self;
}

/*
 * Called instead of initWithStyle if cell comes from a storyboard
 * or nib file.
 *
 * Warning: Will not work if you need to access IBOutlet during custom initialization.
 */
-(id)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self delayContentTouches];
    }
    return self;
}

- (void)delayContentTouches {
    // iterate over all the UITableViewCell's subviews
    for (id view in self.subviews) {
        
        // looking for a UITableViewCellScrollView
        if ([NSStringFromClass([view class]) isEqualToString:@"UITableViewCellScrollView"]) {
            
            // this test is here for safety only, also there is no UITableViewCellScrollView in iOS8
            if([view isKindOfClass:[UIScrollView class]]) {
                
                // turn OFF delaysContentTouches in the hidden subview
                UIScrollView *scroll = (UIScrollView *) view;
                scroll.delaysContentTouches = NO;
            }
            break;
        }
    }
}

@end

File: Patchr/AirTableViewCell.swift
//
//  AirTableViewCell.swift
//  Patchr
//
//  Created by Jay Massena on 10/18/15.
//  Copyright © 2015 3meters. All rights reserved.
//

import UIKit

class AirTableViewCell: UITableViewCell {
	
	var view: BaseView?
	var separator = UIView()
	var padding = UIEdgeInsetsZero
	
	init(view: UIView, padding: UIEdgeInsets, reuseIdentifier: String?) {
		super.init(style: .Default, reuseIdentifier: reuseIdentifier)
		self.view = view as? BaseView
		self.padding = padding
		initialize()
	}
	
	required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	func initialize() {
		addSeparator()
		if self.view != nil {
			self.contentView.addSubview(self.view!)
		}
	}
	
	override func layoutSubviews() {
		
		/* Fill contentView with injected view */
		self.view?.fillSuperviewWithLeftPadding(self.padding.left,
			rightPadding: self.padding.right,
			topPadding: self.padding.top,
			bottomPadding: self.padding.bottom + 1)
		
		self.separator.anchorBottomCenterFillingWidthWithLeftAndRightPadding(0, bottomPadding: 0, height: 1)
	}
	
	func addSeparator() {
		self.separator.layer.backgroundColor = Colors.separatorColor.CGColor
		self.contentView.addSubview(self.separator)
	}
	
	override func prepareForReuse() {
		self.view?.prepareForRecycle()
	}
}
